test/unit/org/apache/cassandra/index/sai/disk/v1/trie/TrieTermsDictionaryTest.java

```diff
@@ -361,6 +363,10 @@ private ByteComparable primaryKey(String token, String partitionKey, String clus
 
     private ByteComparable asByteComparable(String s)
     {
-        return ByteComparable.fixedLength(ByteBufferUtil.bytes(s));
+        // It is important that we use this method and not ByteComparable.fixedLength(ByteBufferUtil.bytes(s))
+        // because they give us different results when comparing in the trie. See the javadoc of fixedLength
+        // for more details, but the tl;dr is that fixedLength only returns correct results for types that are
+        // all the same number of bytes, which is not the case for the strings used in this test.
+        return ByteComparable.of(s);
```

Comment:
The test should work for both and it makes sense to parameterize it to run it both for `of` and `fixedLength(bytes)`.

`fixedLength` is called this way, because when byte-comparables are composed in sequences it is not correct to leave them unencoded unless they always have the same length. If we are using just one value, it is okay to leave it unencoded. Tests of this type usually use `fixedLength` because it is otherwise hard or impossible to construct the situation that needs to be tested (e.g. using `of` "ca" translates to "ca\0" which is not a prefix of "caaaaa\0").

Reply:
Thanks for the clarification @blambov. I looked closer at the usage of these methods for a few hours yesterday afternoon (after making this commit), and your comment helps confirm several details. I'll update the implementation so that it works for both cases.